dvc/path_info.py

```diff
@@ -18,8 +18,24 @@ def overlaps(self, other):
         return self.isin_or_eq(other) or other.isin(self)
 
     def isin_or_eq(self, other):
+        if isinstance(other, list):
```

Comment:
This function should really be left alone, it's meant to specifically compare two `BasePath` instances.

Reply:
You can accomplish this when you actually want to do the comparison with expressions like

```python
if any((self.isin_or_eq(other) for other in others)):
    ...
```

Reply:
Incorporated in https://github.com/iterative/dvc/pull/4938/commits/395167309234a0fb80cd66a416d3b52cc43f2e1e.